File: toolset/jvm/modules/standalone-plugins/plugin/src/main/java/org/example/GreetingPlugin.java
package org.example;

import org.gradle.api.Plugin;
import org.gradle.api.Project;

public class GreetingPlugin implements Plugin<Project> {

    @Override
    public void apply(Project project) {
        // Register the task and apply an action in one step
        project.getTasks().register("hello", task ->
                task.doLast(t ->
                        System.out.println("Hello from the GreetingPlugin")
                )
        );
    }
}

File: toolset/jvm/modules/standalone-plugins/plugin/src/main/kotlin/com/nophasenokill/GreetingPlugin.kt
package com.nophasenokill

import org.gradle.api.Plugin
import org.gradle.api.Project

class GreetingPlugin : Plugin<Project> {
    override fun apply(project: Project) {
        // Register a task
        project.tasks.register("greeting") { task ->
            task.doLast {
                println("Hello from plugin 'com.nophasenokill.greeting'")
            }
        }
    }
}